packages/shared/src/utils/digital-options/digital-options.ts

```diff
@@ -10,27 +10,35 @@ type TShowError = {
     message: string;
     header: string;
     redirect_label: string;
-    redirectOnClick: null;
+    redirectOnClick: () => void | null;
     should_show_refresh: boolean;
     redirect_to: string;
     should_clear_error_on_click: boolean;
+    should_redirect: boolean;
 };
 
 type TAccounts = {
     residence?: string;
     landing_company_shortcode?: string;
 };
 
-export const showDigitalOptionsUnavailableError = (showError: (t: TShowError) => void, message: TMessage) => {
+export const showDigitalOptionsUnavailableError = (
+    showError: (t: TShowError) => void,
+    message: TMessage,
+    redirectOnClick: () => void | null,
+    should_clear_error_on_click: boolean,
+    should_redirect: boolean
+) => {
     const { title, text, link } = message;
     showError({
         message: text,
         header: title,
         redirect_label: link,
-        redirectOnClick: null,
+        redirectOnClick,
         should_show_refresh: false,
         redirect_to: '/mt5',
-        should_clear_error_on_click: true,
+        should_clear_error_on_click: !should_clear_error_on_click ? should_clear_error_on_click : true,
```

Comment:
```suggestion
        should_clear_error_on_click,
```

Reply:
We usually call function `showDigitalOptionsUnavailableError()` without passing `should_clear_error_on_click`, so by default it is `true`. When we don't want to redirect - we should pass `should_clear_error_on_click` as `false`. So current suggestion is not valid...

Reply:
so if !should_clear_error_on_click gives true, we should pass true inside showError, right?
and if !should_clear_error_on_click gives false, we should pass true anyway, right?
so if we pass true all the time, why do we need this change?
it can be:
```suggestion
        should_clear_error_on_click: true,
```

Reply:
so if should_clear_error_on_click is false, we should pass false inside showError, right?
and if should_clear_error_on_click is true, we should pass true, right?
so my suggestion will work, please try to apply it )

Reply:
done✅